### PR TITLE
Migrate to multiple clients and add `convertLead` operation

### PR DESCRIPTION
**Pull request by Chuhaa**
## Purpose

- Migrate existing single client to two clients
- Add `convertLead` operation in SOAP module 

## Goals
Fixes https://github.com/wso2-enterprise/choreo/issues/5319

## Automation tests
 - Unit tests 
Tested
 - Integration tests
Tested

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Samples
```ballerina
import ballerinax/sfdc;
import ballerinax/sfdc.soap;

sfdc:SalesforceConfiguration sfConfig = {
   baseUrl: <"EP_URL">,
   clientConfig: {
     clientId: <"CLIENT_ID">,
     clientSecret: <"CLIENT_SECRET">,
     refreshToken: <"REFRESH_TOKEN">,
     refreshUrl: <"REFRESH_URL"> 
   }
};

soap:Client soapClient = new (sfConfig);
string leadId = "xxxxxxx";
ConvertedLead|error response = soapClient->convertLead(leadId);
```

## Test environment
SLAlpha 5
JDK 11
Ubuntu 20.04

### CODE REVIEW

File: sfdc/rest_client.bal [hunk 1]
```diff
@@ -0,0 +1,527 @@
...
+
+    # Delete existing records based on relevant object id.
+    #
+    # + sObjectName - SObject name value
+    # + id - SObject id
+    # + return - true if successful else false or Error occured
+    @display {label: "Delete Record"}
+    isolated remote function deleteRecord(@display {label: "SObject Name"} string sObjectName, @display 
+                                          {label: "SObject ID"} string id) returns @tainted@display {label: "Result"} 
+    Error? {
```
Comment (sachinira) [inline sfdc/rest_client.bal:160]: Can we change the custom error in these places too? 
Formatting error

**Review by sachinira — CHANGES_REQUESTED** [review]

File: sfdc/modules/bulk/client.bal [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+        return bulkJob;
+    }
+
+    # Get information about a job.
+    #
+    # + bulkJob - Job object of which the info is required 
+    # + return - Job information record or error
+    @display {label: "Get Job Information"}
+    isolated remote function getJobInfo(@display {label: "Bulk Job"} BulkJob bulkJob) returns @tainted@display 
+    {label: "Job Information"} error|JobInfo {
```
[the remark this answers is not in the document]
Comment (Chuhaa) [reply sfdc/modules/bulk/client.bal:111]: This is the formatting provided by ballerina plugin.

File: sfdc/rest_client.bal (hunk 1 above)
Comment (Chuhaa) [reply sfdc/rest_client.bal:160]: This will be a method signature change. Is it possible to do?

Comment (Chuhaa) [reply sfdc/rest_client.bal:160]: @abeykoon please share your thoughts

**Review by abeykoon — APPROVED** [review]

File: sfdc/rest_client.bal (hunk 1 above)
Comment (abeykoon) [reply sfdc/rest_client.bal:160]: As it is a breaking change , we'll not do that.

**Pull request merged**